hibernate-core/src/main/java/org/hibernate/metamodel/mapping/EmbeddableMappingType.java

```diff
@@ -528,7 +528,21 @@ public int forEachJdbcValue(
 		for ( int i = 0; i < attributes.size(); i++ ) {
 			final AttributeMapping attributeMapping = attributes.get( i );
 			final Object o = attributeMapping.getPropertyAccess().getGetter().get( value );
-			span += attributeMapping.forEachJdbcValue( o, clause, span + offset, consumer, session );
+			if ( attributeMapping instanceof ToOneAttributeMapping ) {
```

Comment:
I recently had a SO exception due to something similar. What do you think about changing the implementation of `ToOneAttributeMapping` to traverse only the foreign key by default? I don't know how join fetching is currently implemented so I don't know how such a change would affect that, but I have a feeling that such missing instanceof checks might be hidden problems that we will yet discover.